src/main/java/uk/gov/hmcts/ccd/domain/model/search/elasticsearch/ElasticsearchRequest.java

```diff
@@ -4,39 +4,69 @@
 import com.fasterxml.jackson.databind.node.ArrayNode;
 import com.fasterxml.jackson.databind.node.BooleanNode;
 import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
 import lombok.Data;
 import lombok.NonNull;
 import uk.gov.hmcts.ccd.data.casedetails.search.MetaData;
 
 import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
-import java.util.stream.Collectors;
 import java.util.stream.StreamSupport;
 
+import static java.util.stream.Collectors.toList;
+import static uk.gov.hmcts.ccd.config.JacksonUtils.MAPPER;
+import static uk.gov.hmcts.ccd.data.casedetails.CaseDetailsEntity.DATA_CLASSIFICATION_COL;
+import static uk.gov.hmcts.ccd.data.casedetails.CaseDetailsEntity.DATA_COL;
+
 @Data
 public class ElasticsearchRequest {
 
     public static final String SORT = "sort";
     public static final String SOURCE = "_source";
     public static final String QUERY = "query";
+    public static final String NATIVE_ES_QUERY = "native_es_query";
+    public static final String SUPPLEMENTARY_DATA = "supplementary_data";
     public static final String CASE_DATA_PREFIX = "data.";
+    public static final String SUPPLEMENTARY_DATA_PREFIX = "supplementary_data.";
     public static final String COLLECTION_VALUE_SUFFIX = ".value";
     public static final String WILDCARD = "*";
+    public static final ArrayNode METADATA_FIELDS;
+
+    private JsonNode nativeSearchRequest;
+    private ArrayNode requestedSupplementaryData;
```

Comment:
not 100% sure but would it make the code simpler to have this as an List<String>? so that clients don't have to work with the cumbersome Jakson API

Reply:
Hmm, the `searchRequest` coming into the constructor is already a `JsonNode` that we extract supplementary data out of. Unless we have to start doing any more complex intermediate processing in this class on the object it might be better to keep as is - otherwise it will be coming in as an `ArrayNode`, converting to `List<String>` then converting back to `ArrayNode` to create the "final" request?